src/App.tsx

```diff
@@ -14,11 +14,116 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    data: goodsFromServer,
+    selectedGoods: [],
+  };
+
+  createSelected() {
+    const goods: string[] = this.state.selectedGoods;
+    const { length } = this.state.selectedGoods;
+
+    if (length === 0) {
+      return 'No Selected';
+    }
+
+    if (length === 1) {
+      return `${goods[0]} is selected`;
+    }
+
+    let output = '';
+
+    for (let i = 0; i < length - 1; i += 1) {
+      output += `${goods[i]} , `;
+    }
+
+    return `${output.substring(0, output.length - 2)} and ${goods[goods.length - 1]} are selected`;
+  }
+
+  addWord(word: string) {
+    const arr: string[] = this.state.selectedGoods;
+
+    arr.push(word);
+
+    return arr;
+  }
+
+  removeWord(word: string) {
+    const arr: string[] = this.state.selectedGoods;
+    const index: number = arr.indexOf(word);
+
+    if (index >= 0) {
+      arr.splice(index, 1);
+    }
+
+    return arr;
+  }
+
+  check(word: string) {
+    const arr: string[] = this.state.selectedGoods;
+    const index: number = arr.indexOf(word);
+
+    if (index >= 0) {
+      return true;
+    }
+
+    return false;
+  }
+
+  render() {
+    return (
+      <div className="app">
+        <div className="app__selectedList">
+          <p className="app__selectedList__title">
+            Selected good:
+          </p>
+          <p className="app__selectedList__list">
+            {this.createSelected()}
+          </p>
+          <button
+            type="button"
+            className="app__selectedList__clear"
+            onClick={() => this.setState({ selectedGoods: [] })}
+          >
+            X
+          </button>
+        </div>
+
+        <div className="app__selectedForm">
+          <ul className="app__selectedForm__list">
+            {this.state.data.map(good => (
+              <li className="app__selectedForm__list__item">
+                {`${good}`}
+                {!this.check(good) && (
+                  <button
+                    type="button"
+                    className="app__selectedForm__list__item__add"
+                    onClick={
+                      () => this.setState({ selectedGoods: this.addWord(good) })
```

Comment:
Здесь вроде надо передавать колбек с текущим стейтом в setState
```suggestion
                      () => this.setState((state) => ({ selectedGoods: this.addWord(good, state) }))
```
и внутри addWord уже менять этот стейт, не знаю зачем это нужно конкретно здесь, но в тз требуется

Reply:
так я передаю good в addWord, а addWord  в setState и в setState перезаписывается selectedGoods

...но, я могу и ошибаться(